### PR TITLE
core(dbw response compression): Exclude non binary files from auditing

### PR DESCRIPTION
**Pull request by FadySamirSadek**
I have followed the recommended approach but I am wondering if there is a better way to know the type of the file and if the extensions I added are enough and if not what can I refer to in order to get a full list of extensions.
#4039

### CODE REVIEW

**Comment by sexywife** [comment]
Remove

File: lighthouse-core/gather/gatherers/dobetterweb/response-compression.js [hunk 1]
```diff
@@ -14,28 +14,41 @@ const Gatherer = require('../gatherer');
...
     const unoptimizedResponses = [];
 
     networkRecords.forEach(record => {
       const isTextBasedResource = record.resourceType() && record.resourceType().isTextType();
+      const url = record.url.toLowerCase();
+      let isMediaResource;
+
+      for (let i = 0; i < mediaTypes.length; i+=1) {
+        if (url.endsWith(mediaTypes[i])) {
+          isMediaResource = true;
```
Comment (sexywife) [inline lighthouse-core/gather/gatherers/dobetterweb/response-compression.js]: False

File: lighthouse-core/gather/gatherers/dobetterweb/response-compression.js [hunk 2]
```diff
@@ -14,28 +14,41 @@ const Gatherer = require('../gatherer');
...
       }
 
       const isContentEncoded = record.responseHeaders.find(header =>
-        header.name.toLowerCase() === 'content-encoding' &&
-        compressionTypes.includes(header.value)
+          header.name.toLowerCase() === 'content-encoding' &&
+          compressionTypes.includes(header.value)
       );
 
       if (!isContentEncoded) {
```
Comment (sexywife) [inline lighthouse-core/gather/gatherers/dobetterweb/response-compression.js]: False

Comment (sexywife) [reply lighthouse-core/gather/gatherers/dobetterweb/response-compression.js]: False

File: lighthouse-core/gather/gatherers/dobetterweb/response-compression.js [hunk 3]
```diff
@@ -14,28 +14,41 @@ const Gatherer = require('../gatherer');
...
+          isMediaResource = true;
+          break;
+        }
+      }
       const isChromeExtensionResource = record.url.startsWith(CHROME_EXTENSION_PROTOCOL);
 
       if (!isTextBasedResource || !record.resourceSize || !record.finished ||
-        isChromeExtensionResource || !record.transferSize || record.statusCode === 304) {
+          isChromeExtensionResource || !record.transferSize || record.statusCode === 304
+          ||isMediaResource) {
```
Comment (wardpeet) [inline lighthouse-core/gather/gatherers/dobetterweb/response-compression.js]: nit: add a space between `||isMediaResource` => `|| isMediaResource`

File: lighthouse-core/gather/gatherers/dobetterweb/response-compression.js [hunk 4]
```diff
@@ -14,28 +14,41 @@ const Gatherer = require('../gatherer');
...
-   */
+     * @param {!NetworkRecords} networkRecords
+     * @return {!Array<{url: string, isBase64DataUri: boolean, mimeType: string, resourceSize: number}>}
+     */
   static filterUnoptimizedResponses(networkRecords) {
     const unoptimizedResponses = [];
 
     networkRecords.forEach(record => {
       const isTextBasedResource = record.resourceType() && record.resourceType().isTextType();
+      const url = record.url.toLowerCase();
```
Comment (wardpeet) [inline lighthouse-core/gather/gatherers/dobetterweb/response-compression.js]: `const isMediaResource = (record.resourceType() === WebInspector.resourceTypes.Media || record.resourceType() === WebInspector.resourceTypes.Image;` is probably a better approach.

Although I think we want svg's to be compressed at they are text based and not binary based.

File: lighthouse-core/gather/gatherers/dobetterweb/response-compression.js [hunk 5]
```diff
@@ -14,28 +14,41 @@ const Gatherer = require('../gatherer');
 const gzip = require('zlib').gzip;
 
 const compressionTypes = ['gzip', 'br', 'deflate'];
+const mediaTypes = ['mp3', 'wav', 'wma', 'webm', 'jpg', 'gif', 'png', 'webp',
```
Comment (patrickhulce) [inline lighthouse-core/gather/gatherers/dobetterweb/response-compression.js]: how about we changes this to a set of banned mime types instead of extensions

```js
const binaryMimeTypes = ['image', 'audio', 'video'];
```

File: lighthouse-core/gather/gatherers/dobetterweb/response-compression.js [hunk 6]
```diff
@@ -14,28 +14,41 @@ const Gatherer = require('../gatherer');
...
+  'svg', 'mkv', 'flv', 'gif', 'avi', 'mov', 'mpg', '3gp', 'mpeg', 'avi', 'wmv'];
+
 const CHROME_EXTENSION_PROTOCOL = 'chrome-extension:';
 
 class ResponseCompression extends Gatherer {
   /**
-   * @param {!NetworkRecords} networkRecords
-   * @return {!Array<{url: string, isBase64DataUri: boolean, mimeType: string, resourceSize: number}>}
-   */
+     * @param {!NetworkRecords} networkRecords
```
Comment (patrickhulce) [inline lighthouse-core/gather/gatherers/dobetterweb/response-compression.js]: nit: let's revert these changes to keep it as it was :)

File: lighthouse-core/gather/gatherers/dobetterweb/response-compression.js (hunk 4 above)
Comment (patrickhulce) [reply lighthouse-core/gather/gatherers/dobetterweb/response-compression.js]: in this case we're already being fooled by what Chrome thinks the resource type is unfortunately, let's try to add a mime type sanity check instead :)

File: lighthouse-core/gather/gatherers/dobetterweb/response-compression.js [hunk 7]
```diff
@@ -14,28 +14,41 @@ const Gatherer = require('../gatherer');
...
+     * @return {!Array<{url: string, isBase64DataUri: boolean, mimeType: string, resourceSize: number}>}
+     */
   static filterUnoptimizedResponses(networkRecords) {
     const unoptimizedResponses = [];
 
     networkRecords.forEach(record => {
       const isTextBasedResource = record.resourceType() && record.resourceType().isTextType();
+      const url = record.url.toLowerCase();
+      let isMediaResource;
```
Comment (patrickhulce) [inline lighthouse-core/gather/gatherers/dobetterweb/response-compression.js]: once we have a set of mime types we can combine them into the check for isTextBasedResource since that's what we're really getting at

```js
const isBinaryResource = record.mimeType && binaryMimeTypes.some(mimeType => record.mimeType.startsWith(mimeType));
const isTextBasedResource = !isBinaryResource && record.resourceType() && ...
```

File: lighthouse-core/gather/gatherers/dobetterweb/response-compression.js [hunk 8]
```diff
@@ -14,28 +14,41 @@ const Gatherer = require('../gatherer');
...
-        isChromeExtensionResource || !record.transferSize || record.statusCode === 304) {
+          isChromeExtensionResource || !record.transferSize || record.statusCode === 304
+          ||isMediaResource) {
         return;
       }
 
       const isContentEncoded = record.responseHeaders.find(header =>
-        header.name.toLowerCase() === 'content-encoding' &&
-        compressionTypes.includes(header.value)
+          header.name.toLowerCase() === 'content-encoding' &&
```
Comment (patrickhulce) [inline lighthouse-core/gather/gatherers/dobetterweb/response-compression.js]: nit: let's revert this change too

**Comment by patrickhulce** [comment]
hey @FadySamirSadek thanks for starting this! do you think you have time to carry it across the finish line? 🐎 🏁 

if not no worries, we have another contributor interested in picking this up :)

**Comment by FadySamirSadek** [comment]
@patrickhulce yes will make progress on it today

**Comment by wardpeet** [comment]
@FadySamirSadek any progress? Sorry to keep nagging but it would be awesome if we could land this! 💪

**Comment by FadySamirSadek** [comment]
@wardpeet  @patrickhulce  Thank you for keeping me on this , sorry for making you wait that long for a simple issue. Please provide me with any feedback or changes you request.

File: lighthouse-core/gather/gatherers/dobetterweb/response-compression.js [hunk 9]
```diff
@@ -25,7 +26,10 @@ class ResponseCompression extends Gatherer {
     const unoptimizedResponses = [];
 
     networkRecords.forEach(record => {
-      const isTextBasedResource = record.resourceType() && record.resourceType().isTextType();
+      const isBinaryResource = record.mimeType &&
+      binaryMimeTypes.some(mimeType => record.mimeType.startsWith(mimeType));
```
Comment (patrickhulce) [inline lighthouse-core/gather/gatherers/dobetterweb/response-compression.js]: nit: let's indent this line

File: lighthouse-core/gather/gatherers/dobetterweb/response-compression.js [hunk 10]
```diff
@@ -25,7 +26,10 @@ class ResponseCompression extends Gatherer {
...
+      const isTextBasedResource = !isBinaryResource && record.resourceType() &&
+      record.resourceType().isTextType();
```
Comment (patrickhulce) [inline lighthouse-core/gather/gatherers/dobetterweb/response-compression.js]: nit: this one too :)

**Comment by paulirish** [comment]
feedback addressed. @patrickhulce ptal.
